### PR TITLE
Add Users page

### PR DESCRIPTION
**Pull request by ssaitejar96**
Includes:
- Dropdown to select Membership type
- Table showing Address,Number of passports owned

### CODE REVIEW

**Comment by ssaitejar96** [comment]
cc'ing @dvargas92495 for review.

File: packages/app/components/UsersTabContent.tsx [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+            setShowLoading(true);
+            const contract = new web3.eth.Contract(getAbiFromJson(passportJson));
+            contract.options.address = selectedOption;
+            // Get tokenIds which have been bought
+            contract.getPastEvents('Transfer', {
+                filter: {
+                    _from: '0x0000000000000000000000000000000000000000'
+                },
+                fromBlock: 0
+            }).then((events) => {
```
Comment (dvargasfuertes) [inline packages/app/components/UsersTabContent.tsx:87]: ooh nice! wasn't aware of this method

File: packages/app/components/UsersTabContent.tsx [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+                        symbol: p[1],
+                        supply: p[2],
+                        price: web3.utils.fromWei(p[3], "ether"),
+                    };
+                    return data;
+                });
+            });
+            Promise.all(promises).then((values) => {
+                const data = Object.assign({}, ...values);
+                setMembershipDetails(data);
```
Comment (dvargasfuertes) [inline packages/app/components/UsersTabContent.tsx:127]: nit: why not just do `setMembershipDetails(values)`?

File: packages/app/components/UsersTabContent.tsx [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+                        owners[ownerAddr] = (owners[ownerAddr]+1) || 1;
+                    });
+                    setUsers(owners);
+                })
+                .finally(() => setShowLoading(false));
+            });
+        }
+    }, [selectedOption, setUsers]);
+
+    useEffect(() => {
```
Comment (dvargasfuertes) [inline packages/app/components/UsersTabContent.tsx:107]: nit: It looks like we could combine this `useEffect` with the first one, save a render

File: packages/app/pages/checkout/[network]/[address].tsx [hunk 4]
```diff
@@ -140,7 +140,7 @@ const CheckoutPage = ({
     axios.post("/api/redirectionUrl", {
       address: address
     })
-    .then((result) => {
+    .then((result: { data: { redirect_url: string } }) => {
```
Comment (dvargasfuertes) [inline packages/app/pages/checkout/[network]/[address].tsx:143]: nit: axios has a type signature `axios.post<{redirect_url:string}>`

File: packages/app/pages/index.tsx [hunk 5]
```diff
@@ -23,6 +23,7 @@ import {
   Web3Provider,
 } from "../components/Web3Context";
 import axios from "axios";
+import UsersTabContent from "../components/UsersTabContent";
```
Comment (dvargasfuertes) [inline packages/app/pages/index.tsx:26]: I can move the Memberships Tab Content into its own file as well in my PR

**Review by dvargasfuertes — APPROVED** [review]
Good stuff! Going to deploy it on the branch and poke around later today, then merge once I verify on the branch

File: packages/app/components/UsersTabContent.tsx [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+                    />
+                    {Object.keys(users).length>0 ?
+                        (<Table>
+                            <tr>
+                                <TableHeader>Address</TableHeader>
+                                <TableHeader>Passports owned</TableHeader>
+                            </tr>
+                            {
+                                Object.keys(users).map((user) => {
+                                    return (<tr>
```
Comment (dvargasfuertes) [inline packages/app/components/UsersTabContent.tsx]: Deployment failed here, will need to address this first

image

File: packages/app/components/UsersTabContent.tsx (hunk 2 above)
Comment (ssaitejar96) [reply packages/app/components/UsersTabContent.tsx:127]: values is an array, data is an object.

File: packages/app/components/UsersTabContent.tsx (hunk 6 above)
Comment (ssaitejar96) [reply packages/app/components/UsersTabContent.tsx]: My bad....forgot to run the linter. Fixed now.

**Comment by ssaitejar96** [comment]
@dvargas92495 fixed the lint issue now

File: packages/app/components/UsersTabContent.tsx [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+                    const owners: {[key: string]: number} = {};
+                    ownerAddrs.forEach((ownerAddr) => {
+                        owners[ownerAddr] = (owners[ownerAddr]+1) || 1;
+                    });
+                    setUsers(owners);
+                })
+                .finally(() => setShowLoading(false));
+            });
+        }
+    }, [selectedOption, setUsers]);
```
Comment (dvargasfuertes) [inline packages/app/components/UsersTabContent.tsx:105]: image

Seems like there was another lint warning and a swc failure. I'm not sure why the swc failure would happen, but could we fix these lint warnings and see it's related?

**Comment by ssaitejar96** [comment]
@dvargas92495  npm run build actually ran fine on my local terminal even with the warnings. I regenerated package-lock.json and pushed to see if it makes a difference as suggested [here](https://stackoverflow.com/a/70360521)

According to https://nextjs.org/docs/messages/failed-loading-swc,
Screenshot 2022-01-30 at 9 08 19 AM

Could you try the first option if the new package-lock also doesnt work?

**Comment by dvargasfuertes** [comment]
This is dope! merging

image

**Pull request merged**